### PR TITLE
Add some basic e2e tests

### LINKED ISSUE #434: Full e2e test that builds a single and multi cluster on AWS
Go through all vital steps of tarmak:

- Start with `tarmak init`
- `tarmak cluster image build`
- `tarmak cluster apply` 
-  sonobouy maybe 
- `tarmak cluster destroy`

The build of tarmak should use a bundled wing and ensure that this is used during cluster operations

### PR DESCRIPTION
**Pull request by JoshVanL**
/assign @simonswine 
fixes #434 

```release-note
Create end to end tests in code
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -291,3 +291,21 @@ docker_%:
 
 	# remove container
 	docker rm $(CONTAINER_ID)
+
+e2e-test-all-conformance: build
```
Comment (simonswine) [inline Makefile]: Can you add some help text comment so it shows up in `make help`?

File: cmd/tarmak/e2e/sonobuoy_test.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+		t:       t,
+		binPath: "../../../bin/sonobuoy",
+		mode:    mode,
+		retrievePath: fmt.Sprintf("%s/%s-test-%s",
+			os.TempDir(), environment, randStringRunes(6)),
+		kubeconfig: kubeconfig,
+	}
+
+	if _, err := os.Stat(si.binPath); os.IsNotExist(err) {
+		t.Fatal("sonobuoy binary not exissing: ", si.binPath)
```
Comment (simonswine) [inline cmd/tarmak/e2e/sonobuoy_test.go]: existing

File: cmd/tarmak/e2e/sonobuoy_test.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+	binPath      string // bin path to tarmak binary
+	mode         string // quick or conformance
+	retrievePath string
+	kubeconfig   string
+}
+
+func NewSonobuoyInstance(t *testing.T, environment, kubeconfig string) *SonobuoyInstance {
+	si := &SonobuoyInstance{
+		t:       t,
+		binPath: "../../../bin/sonobuoy",
```
Comment (simonswine) [inline cmd/tarmak/e2e/sonobuoy_test.go]: Not too sure where is this supposed to come from?

Did you want to add it to the Makefile

File: packer/amazon/centos-puppet-agent-latest-kernel.json [hunk 4]
```diff
@@ -26,7 +26,7 @@
       "ssh_username": "centos",
       "ssh_pty": "true",
       "encrypt_boot": "{{user `ebs_volume_encrypted`}}",
-      "ami_name": "Tarmak CentOS 7 x86_64 with puppet-agent and latest upstream kernel{{isotime \"2006-01-02_030405\"}}",
+      "ami_name": "{{user `tarmak_environment`}} - Tarmak CentOS 7 x86_64 with puppet-agent and latest upstream kernel{{isotime \"2006-01-02_030405\"}}",
```
Comment (simonswine) [inline packer/amazon/centos-puppet-agent-latest-kernel.json]: Is that change necessary?

File: packer/amazon/centos-puppet-agent.json [hunk 5]
```diff
@@ -26,7 +26,7 @@
       "ssh_username": "centos",
       "ssh_pty": "true",
       "encrypt_boot": "{{user `ebs_volume_encrypted`}}",
-      "ami_name": "Tarmak CentOS 7 x86_64 with puppet-agent {{isotime \"2006-01-02_030405\"}}",
+      "ami_name": "{{user `tarmak_environment`}} - Tarmak CentOS 7 x86_64 with puppet-agent {{isotime \"2006-01-02_030405\"}}",
```
Comment (simonswine) [inline packer/amazon/centos-puppet-agent.json]: Is that change necessary?

File: puppet/modules/kubernetes/manifests/dns.pp [hunk 6]
```diff
@@ -31,18 +31,18 @@
   $post_1_10 = versioncmp($::kubernetes::version, '1.10.0') >= 0
 
   if $post_1_10 {
-    $app_name = 'core-dns'
```
Comment (simonswine) [inline puppet/modules/kubernetes/manifests/dns.pp]: Is that name change down to sonobouy?

Comment (JoshVanL) [reply puppet/modules/kubernetes/manifests/dns.pp]: Yes, guess we should consider it source of truth

File: packer/amazon/centos-puppet-agent.json (hunk 5 above)
Comment (JoshVanL) [reply packer/amazon/centos-puppet-agent.json]: No, change came from rebase, will revert

File: packer/amazon/centos-puppet-agent-latest-kernel.json (hunk 4 above)
Comment (JoshVanL) [reply packer/amazon/centos-puppet-agent-latest-kernel.json]: No, change came from rebase, will revert

File: puppet/modules/kubernetes/manifests/dns.pp (hunk 6 above)
Comment (simonswine) [reply puppet/modules/kubernetes/manifests/dns.pp]: I am sure kube-dns is the correct name for it though
